### PR TITLE
Add possibility to verify execution statistics

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: 1.0/gradle/vividus-test.gradle [hunk 1]
```diff
@@ -120,6 +120,21 @@ class RunStoriesTask extends JavaExec {
...
 
+def expectedResult = project.findProperty('validateRunStatistics');
+if (expectedResult) {
+    tasks.register('verifyStatistics') {
+    dependsOn 'runStories'
+        doLast {
+            File actualFile = new File(project.buildDir, 'statistics/statistics.json')
+            File expectedFile = new File(project.buildDir, expectedResult)
+            def actualMap = new groovy.json.JsonSlurper().parseText(actualFile.text)
+            def expectedMap = new groovy.json.JsonSlurper().parseText(expectedFile.text)
```
Comment (uarlouski) [inline 1.0/gradle/vividus-test.gradle:131]: groovy.json.JsonSlurper() reuse?)

File: 1.0/gradle/vividus-test.gradle [hunk 2]
```diff
@@ -120,6 +120,21 @@ class RunStoriesTask extends JavaExec {
...
+            logger.warn("Expected execution statistics:\n {} actual ecxecution statistics :\n {}", expectedFile.text, actualFile.text)
```
Comment (uarlouski) [inline 1.0/gradle/vividus-test.gradle:132]: does it support lazy logging?

File: 1.0/gradle/vividus-test.gradle [hunk 3]
```diff
@@ -132,14 +132,36 @@ task runStories (dependsOn: build, type: RunStoriesTask, group: 'Vividus') {
...
+        }
+        else if (treatKnownIssuesOnlyAsPassed && executionResult.get().getExitValue() == 1) {
             logger.warn('\nKNOWN ISSUES ONLY: All test failures are known issues')
         } else {
             executionResult.get().assertNormalExitValue()
         }
     }
 }
 
+tasks.register('verifyRunStatistics') {
```
Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:147]: why not to register it in a regular way?

File: 1.0/gradle/vividus-test.gradle [hunk 4]
```diff
@@ -132,14 +132,36 @@ task runStories (dependsOn: build, type: RunStoriesTask, group: 'Vividus') {
...
+    def expectedResult = project.findProperty(property)
+    if (!expectedResult) {
+        throw new StopExecutionException("project property \"$property\" should be set")
+    }
+    doLast {
+        File actualFile = new File(project.buildDir, 'statistics/statistics.json')
+        File expectedFile = new File(project.buildDir, expectedResult)
+        def actualMap = new groovy.json.JsonSlurper().parseText(actualFile.text)
+        def expectedMap = new groovy.json.JsonSlurper().parseText(expectedFile.text)
+        logger.warn("Expected execution statistics:\n {} actual ecxecution statistics :\n {}", expectedFile.text, actualFile.text)
```
Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:159]: why `warn`?

File: 1.0/gradle/vividus-test.gradle (hunk 3 above)
Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:147]: `validateRunStatistics `

**Review by valfirst — CHANGES_REQUESTED** [review]

File: 1.0/gradle/vividus-test.gradle (hunk 3 above)
Comment (ikalinin1) [reply 1.0/gradle/vividus-test.gradle:147]: to avoid parameter validation during the configuration phase when task is not used

Comment (ikalinin1) [reply 1.0/gradle/vividus-test.gradle:147]: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:how_do_i_defer_creation

File: 1.0/gradle/vividus-test.gradle (hunk 4 above)
Comment (ikalinin1) [reply 1.0/gradle/vividus-test.gradle:159]: To print it explicitly

**Review by valfirst — APPROVED** [review]

**Pull request merged**